payloads/start.go

```diff
@@ -242,6 +242,10 @@ type StartCmd struct {
 	// Storage contains all the information required to attach or boot
 	// from storage for the new instance.
 	Storage []StorageResource `yaml:"storage,omitempty"`
+
+	// Migration is set to true if the payload represents a request to
+	// restart an existing instance on a new node.
+	Migration bool
```

Comment:
@kaccardi What do you think about this proposal? I did things this way as it seemed more in line with controller's design, in that it relies on launcher to provide information about the state of the instances, although in some [ways](https://github.com/01org/ciao/pull/1226/files#diff-8d268b1ee13fa436f60ebc3d2d1ce9cfR1115) I'm already breaking the existing model by explicitly setting the state of instances inside controller.

But we could add new instance states, e.g., stopping, and set this in controller.  This would simplify launcher but would require more changes to controller and changes to the way we process stats and may break other things.

If you would prefer this approach I can investigate to see how hard this would be.  It may make sense to do this as a separate PR, in which we introduce other states such as deleting, move the instance status from temporary storage, check the stats for missing instances (in case we missed delete events, etc.)